File: bindings/swift/examples/epoch.swift
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import Foundation
import IotaSDK

@main
struct EpochExample {
  static func main() async throws {
    let client = GraphQlClient.newTestnet()

    // Get current epoch
    guard let currentEpoch = try await client.epoch() else {
      throw NSError(
        domain: "Epoch", code: 1,
        userInfo: [NSLocalizedDescriptionKey: "missing current epoch"])
    }

    print("Current epoch: \(currentEpoch.epochId)")
    print("Current epoch start time: \(currentEpoch.startTimestamp)")

    // Get previous epoch
    let previousEpochId = currentEpoch.epochId - 1
    guard let previousEpoch = try await client.epoch(epoch: previousEpochId) else {
      throw NSError(
        domain: "Epoch", code: 1,
        userInfo: [NSLocalizedDescriptionKey: "missing previous epoch"])
    }

    print("Previous epoch: \(previousEpoch.epochId)")
    if let totalStakeRewards = previousEpoch.totalStakeRewards {
      print("Previous epoch stake rewards: \(totalStakeRewards)")
    }
  }
}

File: bindings/typescript/examples/epoch.ts
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { GraphQlClient } from "../lib";

async function main() {
  const client = GraphQlClient.newTestnet();

  // Get current epoch
  const currentEpoch = await client.epoch();
  if (currentEpoch === undefined) {
    throw new Error("missing current epoch");
  }

  console.log(`Current epoch: ${currentEpoch.epochId}`);
  console.log(`Current epoch start time: ${currentEpoch.startTimestamp}`);

  // Get previous epoch
  const previousEpochId = currentEpoch.epochId - 1n;
  const previousEpoch = await client.epoch(previousEpochId);
  if (previousEpoch === undefined) {
    throw new Error("missing previous epoch");
  }

  console.log(`Previous epoch: ${previousEpoch.epochId}`);
  if (previousEpoch.totalStakeRewards !== undefined) {
    console.log(`Previous epoch stake rewards: ${previousEpoch.totalStakeRewards}`);
  }
}

main();
